public/components/datasources/components/manage/accelerations/create_accelerations_flyout/selectors/preview_sql_defintion.tsx

```diff
@@ -26,25 +32,89 @@ export const PreviewSQLDefinition = ({
   accelerationFormData,
   setAccelerationFormData,
 }: PreviewSQLDefinitionProps) => {
+  const { setToast } = useToast();
   const [isPreviewStale, setIsPreviewStale] = useState(false);
   const [isPreviewTriggered, setIsPreviewTriggered] = useState(false);
   const [sqlCode, setSQLcode] = useState('');
+  const [sqlWorkbenchPLuginExists, setSQLWorkbenchPluginExists] = useState(false);
 
-  const onClickPreview = () => {
+  const checkForErrors = () => {
     const errors = formValidator(accelerationFormData);
     if (hasError(errors)) {
       setAccelerationFormData({ ...accelerationFormData, formErrors: errors });
+      return true;
+    } else return false;
+  };
+
+  const onClickPreview = () => {
+    if (checkForErrors()) {
       return;
     }
     setSQLcode(accelerationQueryBuilder(accelerationFormData));
     setIsPreviewStale(false);
     setIsPreviewTriggered(true);
   };
 
+  const checkIfSQLWorkbenchPluginIsInstalled = () => {
+    fetch('../api/status', {
+      headers: {
+        'Content-Type': 'application/json',
+        'osd-xsrf': 'true',
+        'accept-language': 'en-US,en;q=0.9,zh-CN;q=0.8,zh;q=0.7,zh-TW;q=0.6',
```

Comment:
just for my knowledge why do we need this in the header, isn't this for the `.i18nrc`?

Reply:
that's a header added to request to accept responses in different languages